swap/src/bitcoin/wallet.rs

```diff
@@ -202,106 +326,79 @@ impl Wallet {
-            })?;
+        loop {
+            let status = self.client.lock().await.status_of_script(&script, &txid)?;
 
-            Result::<_, backoff::Error<Error>>::Ok(tx)
-        })
-        .await
-        .context("Transient errors should be retried")?;
+            tracing::debug!("Status of script {} is {}", script, status);
 
-        Ok(tx)
+            if status_fn(status) {
+                break;
+            }
+
+            tokio::time::sleep(Duration::from_secs(5)).await;
+        }
+
+        // TODO: Unsubscribe using the client? Or at least clear our local data, we
+        // should never get a notification again.
+
+        Ok(())
     }
 
     pub async fn get_block_height(&self) -> Result<BlockHeight> {
-        let url = make_blocks_tip_height_url(&self.http_url)?;
-
-        let height = retry(ConstantBackoff::new(Duration::from_secs(1)), || async {
-            let height = reqwest::get(url.clone())
-                .await
-                .map_err(Error::Io)?
-                .text()
-                .await
-                .map_err(Error::Io)?
-                .parse::<u32>()
-                .map_err(|err| backoff::Error::Permanent(Error::Parse(err)))?;
-            Result::<_, backoff::Error<Error>>::Ok(height)
-        })
-        .await
-        .context("Transient errors should be retried")?;
+        let mut inner = self.client.lock().await;
 
-        Ok(BlockHeight::new(height))
+        inner.drain_notifications()?;
+
+        Ok(inner.latest_block)
     }
 
-    pub async fn transaction_block_height(&self, txid: Txid) -> Result<BlockHeight> {
+    pub async fn transaction_block_height(&self, txid: Txid) -> Result<Option<BlockHeight>> {
         let status_url = make_tx_status_url(&self.http_url, txid)?;
 
         #[derive(Serialize, Deserialize, Debug, Clone)]
         struct TransactionStatus {
             block_height: Option<u32>,
             confirmed: bool,
         }
-        let height = retry(ConstantBackoff::new(Duration::from_secs(1)), || async {
-            let block_height = reqwest::get(status_url.clone())
-                .await
-                .map_err(|err| backoff::Error::Transient(Error::Io(err)))?
-                .json::<TransactionStatus>()
-                .await
-                .map_err(|err| backoff::Error::Permanent(Error::JsonDeserialization(err)))?
-                .block_height
-                .ok_or(backoff::Error::Transient(Error::NotYetMined))?;
-
-            Result::<_, backoff::Error<Error>>::Ok(block_height)
-        })
-        .await
-        .context("Transient errors should be retried")?;
 
-        Ok(BlockHeight::new(height))
+        let block_height = reqwest::get(status_url.clone())
+            .await
+            .context("Failed to send request")?
+            .json::<TransactionStatus>()
+            .await
+            .context("Failed to deserialize response as TransactionStatus")?
+            .block_height;
```

Comment:
Should we actually remove `retry` from this one?

Reply:
I will try and replace transaction block height with script status and this can go